### PR TITLE
Correct the selected values rendering at isMulti SelectWidget

### PR DESCRIPTION
**Pull request by ionlizarazu**
move isObject condition below isArray condition for a correct multi-select selected values rendering.

See: https://github.com/plone/volto/pull/1916#issuecomment-823415191

### CODE REVIEW

**Review by sneridagh — APPROVED** [review]
Hey @ionlizarazu , I vaguely remember that this had a reason (see https://github.com/plone/volto/commit/176059f82190c034b88927ba59afb536789f7cfb), at some point in some edge case, Plone returns an object (instead of an array). Might be because of fixed (choice) arrays?

My fault for not providing a proper test to catch this up... Anyways, the code seems fine since we test for all the use cases.

Maybe you have another edge case in your hands right now. If you have time, could you add a quick unit test for it? Thanks!!

**Comment by ionlizarazu** [comment]
I'm trying to add a multi-selectable field in one of my custom Blocks.
I add this code  in schema.properties rendering it as multi-select field:

```
export const CardContainerSchema = (types) => ({
[...]
properties: {
    contentTypes: {
        title: 'Content Types',
        choices: types,
        isMulti: true,
    }
```

The `types` parameter is the object composed by the contentType values in the state which are modified for a correct rendering of the choices in my Block Edit.jsx file:
```
const state_types = useSelector(state => state.types.types); 
const types = state_types.map((type) => ([type.title, type.title]))
```
This is the behaviour of my block before the modification :
selectMultiple not working

After the pull request:
selectMultiple after modifications

**Comment by tiberiuichim** [comment]
@ionlizarazu May I suggest using the Listing block? We need to modernize the Listing block, perhaps it's better if you rewrite some of the listing block innards to use BlockDataForm and then add a new listing template to that.

File: src/components/manage/Widgets/SelectWidget.jsx [hunk 1]
```diff
@@ -96,17 +96,17 @@ function getDefaultValues(choices, value) {
...
+
   if (isArray(value) && choices.length > 0) {
     return value.map((v) => ({
       label: find(choices, (o) => o[0] === v)?.[1] || v,
       value: v,
     }));
+  } else if (isObject(value)) {
+    return {
+      label: value.title !== 'None' && value.title ? value.title : value.token,
+      value: value.token,
```
Comment (nileshgulia1) [inline src/components/manage/Widgets/SelectWidget.jsx:108]: Notice that choices behave differently when comes as an object (`value.token`), I think that's what Victor was talking about!
Nonetheless, I'm okay with the schema you mentioned as that gonna work with Array of objects passed.

**Review by nileshgulia1 — APPROVED** [review]
LGTM when it comes to handle multiple values with the given schema!

**Comment by ionlizarazu** [comment]
> @ionlizarazu May I suggest using the Listing block? We need to modernize the Listing block, perhaps it's better if you rewrite some of the listing block innards to use BlockDataForm and then add a new listing template to that.

Nice suggestion, we will look into it, even so, with this pull request we want to get the expected functionality of a multi-select field working, using the schema configuration.

**Review by tiberiuichim — APPROVED** [review]
